internal/node/node.go

```diff
@@ -13,6 +17,7 @@ type Node interface {
 	GetNodesListBySelector(ctx context.Context, selector map[string]string) ([]v1.Node, error)
 	GetNumTargetedNodes(ctx context.Context, selector map[string]string) (int, error)
 	FindNodeCondition(cond []v1.NodeCondition, conditionType v1.NodeConditionType) *v1.NodeCondition
+	UpdateNodeLabels(ctx context.Context, deprecatedNodeModuleReadyLabels sets.Set[string], loaded []types.NamespacedName, unloaded []types.NamespacedName, node *v1.Node) error
```

Comment:
```suggestion
	UpdateLabels(ctx context.Context, node *v1.Node, toBeAdded, toBeRemoved []string) error
```

Few comments here:
1. Everything in this package is related to nodes, therefore, there the name `UpdateNodeLabels` can just be `UpdateLabels`.
2. The main object here is the node, therefore, it should be the first param in the method
3. This methods know nothing about `modules`, `readyLabels` etc - it should be simple - get a slice of labels to be added and a list of labels to be removed and just add/remove them from the node
4. Since you are having an API call to the kubeAPI for each label (`meta.RemoveLabel` one by one), I would split this method into 2 `AddLabels` and `RemoveLabels`. If you do a single API call to the kubeAPI for adding/removing all the laels in a single call then it make sense to have a single method for it.

Reply:
understood, this is fixed.
now before `UpdateLabels` is being called, we create those two string slices out of the `type.NamespacedName` slices and the set